Adding rules

Chiamakabrowneyes opened the pull request:
## Description

Adding the rules component: 
Demo --> https://www.loom.com/share/b0b247a4fcca4027826011bd340af077 

## Link to Ticket: 

Ticket: https://www.notion.so/bathientran/Rules-Page-181977aa180e4cdcbcfb04c5cf314117?pvs=4



## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## What Has Changed

Please describe in bullets what you have changed.

- Add rule section: 
- Added new route app/rules
- Allows users to edit and create new rules

## How Has This Been Tested / How Can This Be Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

Run the dev page and ensure that theres no bugs. Should be able to see the UI for the rules in "app/rules" route.
![image](https://github.com/minerva-startup-spring2024/roommate-organizer/assets/89539046/00f855a8-8320-46b8-bab7-9dfd85d27bac)

Made it adaptable to mobile:
![image](https://github.com/minerva-startup-spring2024/roommate-organizer/assets/89539046/4e18be4f-dba4-40b6-99fe-9d90ba37b6b1)

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
Merge branch 'main' of github.com:minerva-startup-spring2024/roommate-organizer into chiamaka
s	# the commit.
This commit is to upload template changes for house rules, etc

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
merge conflicts

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
Merge branch 'main' of github.com:minerva-startup-spring2024/roommate-organizer into chiamaka
merging to fix database error and work # the commit.

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

vickipetrova commented:
@Chiamakabrowneyes you should fix the deployment -- there's a check that fails, if we merge it rn it will break the deployment.

vickipetrova commented:
> @Chiamakabrowneyes you should fix the deployment -- there's a check that fails, if we merge it rn it will break the deployment.

Actually, the break wasn't from this PR. It was from a failed migration directly applied to the production database. I fixed that issue so now all checks go through. Let's leave this PR open for now. As we discussed in the retrospective today, the rules feature is not part of the MVP and we are prioritizing the MVP tasks for the next two weeks. If we have time and users request this type of feature we can extend it so that -- there's a way to see the rules, add them, edit them, delete them, etc.

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

bathientran committed:
chore(announcements): changes how announcements are displayed and adds view all option

bathientran committed:
chore(announcements): refactors announcement implementation

bathientran committed:
chore(announcements): refactors announcement implementation

Commit:
conflict..

Commit:
conflict..

minerva-startup-spring2024 committed:
Merge pull request #60 from minerva-startup-spring2024/staging

DEPLOY 0.0.1

Chiamakabrowneyes commented:
@vickipetrova @bathientran. 
Since we are done with the main MVP features. I've worked on the the backend functionailty for this rules component. 
![image](https://github.com/minerva-startup-spring2024/roommate-organizer/assets/89539046/fdf15f26-24f6-4bac-95ad-eae3917d2264)
![image](https://github.com/minerva-startup-spring2024/roommate-organizer/assets/89539046/adc6a683-fb00-4a47-a2a9-1708c99022a8)

Chiamakabrowneyes committed:
Merge branch 'main' of github.com:minerva-startup-spring2024/roommate-organizer into adding_ruls

Chiamakabrowneyes committed:
adding backend for rules

Chiamakabrowneyes committed:
adding backend for rules

Chiamakabrowneyes committed:
fixing error

Chiamakabrowneyes committed:
fixing error

Chiamakabrowneyes committed:
added forgot password functionality

Chiamakabrowneyes committed:
added forgot password functionality

Chiamakabrowneyes committed:
making changes to view

Chiamakabrowneyes committed:
making changes to view

Chiamakabrowneyes committed:
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into forgot-password

Chiamakabrowneyes committed:
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
merging most recent changes to mobile view
Merging current changes for rules in mobile viewbranch 'forgot-password' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
adapting the mobile view for rules function

Chiamakabrowneyes committed:
adapting the mobile view for rules function

Chiamakabrowneyes committed:
adding mobile func

Chiamakabrowneyes committed:
adding mobile func

Chiamakabrowneyes committed:
attempting to fix migration error by deleting migration folder locally

Chiamakabrowneyes committed:
attempting to fix migration error by deleting migration folder locally

Chiamakabrowneyes committed:
cleaning up the unneccessary code

Chiamakabrowneyes committed:
cleaning up the unneccessary code

bathientran committed:
chore(): adds migrations back

bathientran commented on src/app/_components/Login.js:
Have you tried testing this change? We don't actually use this component anymore.

Chiamakabrowneyes replied:
Hey yeah, I tested it by running the app locally. It seems to work fine. 
I was reversing the changes that I had made, and that one seemed to pop up so I took it back to what I thought was its original version. 
The rules component doesn't directly use it though.

bathientran replied:
If it isn't part of the rules feature, do you mind removing it then? makes our PRs cleaner and doesn't add tech debt to our repo

Chiamakabrowneyes replied:
ALright on it

Chiamakabrowneyes replied:
Okay done, Can you add back the migrations. Cause I had deleted it locally (and I saw that you added in the initial commit)

Chiamakabrowneyes committed:
adding updates to branch
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
cleaning up the unneccessary code

Chiamakabrowneyes committed:
cleaning up the unneccessary code

bathientran commented:
Also you shouldn't be deleting the migrations, this would break our app, I fixed it in a previous commit. Could you add those back as well please? Thanks

bathientran commented:
Same for the local .env file don't delete that file

Chiamakabrowneyes committed:
adding back the migration folder

Chiamakabrowneyes commented:
I didnt delete the env file. Both of them are still present locally. idk why it shows as deleted. 

But yeah, I've added back the migrations. I had initially deleted it while trying to resolve the database error from earl;ier today

bathientran committed:
chore(): adds back the .env.local file
